dev-core

```diff
@@ -1,16 +1,49 @@
 #!/bin/bash
 
-echo "Building mesg/core:local..."
-docker build -t mesg/core:local --build-arg version="local" .
+set -e
+
+DIR=${GOPATH:-$HOME/go}
+VERSION=local
+LDFLAGS="-X 'github.com/mesg-foundation/core/version.Version=$VERSION'"
+
+echo "compile cli and core"
+GOOS=linux GOARCH=amd64 go build -o ./bin/core -ldflags="$LDFLAGS" core/main.go
+go build -o ./bin/cli -ldflags="$LDFLAGS" interface/cli/main.go
+
+CORE_SUM_PATH="./bin/.core.sum"
+touch "$CORE_SUM_PATH"
+
+# check if core bin was is cached
+CORE_SUM="$(openssl md5 ./bin/core)"
+CORE_SUM_PREV="$(cat $CORE_SUM_PATH)"
+if [[ "$CORE_SUM" == "$CORE_SUM_PREV" ]]; then
+  BINCACHED=true
+else
+  echo "$CORE_SUM" > "$CORE_SUM_PATH"
+fi
 
 # create a symlink for system services.
-rm -rf ~/.mesg/systemservices
-mkdir -p ~/.mesg
-ln -s $GOPATH/src/github.com/mesg-foundation/core/systemservices/sources ~/.mesg/systemservices
+rm -rf "$HOME/.mesg/systemservices"
+mkdir -p "$HOME/.mesg/"
+ln -s "$DIR/src/github.com/mesg-foundation/core/systemservices/sources" "$HOME/.mesg/systemservices"
+
+echo "build mesg/core image"
+if [[ ! $BINCACHED ]]; then
```

Comment:
I don't quite get why are we doing our own cache check? Previous version of _dev-core_ already had Docker cache and if source code is not changed it doesn't even compile the source code use cache instead. What is the thing that we're making faster?

Reply:
```bash
$ cat Dockerfile.dev
FROM ubuntu:18.04
COPY ./bin/core .
CMD [ "./core" ]

$ time docker build -f Dockerfile.dev -t "mesg/core:local" . > /dev/null
docker build -f Dockerfile.dev -t "mesg/core:local" . > /dev/null  2,98s user 0,61s system 85% cpu 4,224 total

$ # ~3 seconds to copy and build the image

$ time openssl md5 ./bin/core > /dev/null
openssl md5 ./bin/core  0,04s user 0,01s system 95% cpu 0,056 total

$ # compared to ~0s to check if core binary has changed - if not then don't the image.
```

Reply:
So go build also uses go cache and if the source hasn't changed it won't rebuild it (it wasn't possible when we build inside docker build image) so this is first step to make it faster. The second one is to avoid build mesg:core/local if core md5 sum hasn't changed.